servers/store/src/main/java/org/projectnessie/server/store/TableCommitMetaStoreWorker.java

```diff
@@ -111,15 +115,18 @@ public Content valueFromStore(ByteString onReferenceValue, Optional<ByteString>
         }
         return builder.build();
 
-      case ICEBERG_METADATA_POINTER:
+      case ICEBERG_REF_STATE:
         ObjectTypes.Content global =
-            globalContent.orElseThrow(TableCommitMetaStoreWorker::noIcebergGlobal);
-        if (!global.hasIcebergGlobal()) {
-          throw noIcebergGlobal();
+            globalContent.orElseThrow(TableCommitMetaStoreWorker::noIcebergMetadataPointer);
+        if (!global.hasIcebergMetadataPointer()) {
+          throw noIcebergMetadataPointer();
         }
         return IcebergTable.of(
-            content.getIcebergMetadataPointer().getMetadataLocation(),
-            global.getIcebergGlobal().getIdGenerators(),
+            global.getIcebergMetadataPointer().getMetadataLocation(),
+            content.getIcebergRefState().getSnapshotId(),
+            content.getIcebergRefState().getSchemaId(),
+            content.getIcebergRefState().getSpecId(),
+            content.getIcebergRefState().getSortOrderId(),
             content.getId());
```

Comment:
nit: might using a builder be preferable here, to reduce the chance of mismatch in ID args, which have the same java type?

Reply:
addressed